examples/tutorials/nb_python/managed_rigid_object_tutorial.py

```diff
@@ -0,0 +1,560 @@
+    if make_video and not os.path.exists(output_path):
+        os.mkdir(output_path)
+
+    # %%
+    # [initialize]
+    # create the simulators AND resets the simulator
+
+    cfg = make_configuration()
+    try:  # Got to make initialization idiot proof
+        sim.close()
+    except NameError:
+        pass
+    sim = habitat_sim.Simulator(cfg)
+    agent_transform = place_agent(sim)
+
+    # get the primitive assets attributes manager
+    prim_templates_mgr = sim.get_asset_template_manager()
+
+    # get the physics object attributes manager
+    obj_templates_mgr = sim.get_object_template_manager()
+
+    # get the rigid object manager
+    rigid_obj_mgr = sim.get_rigid_object_manager()
+    # [/initialize]
+
+    # %%
+    # [basics]
+
+    # load some object templates from configuration files
+    sphere_template_id = obj_templates_mgr.load_configs(
+        str(os.path.join(data_path, "test_assets/objects/sphere"))
+    )[0]
+
+    # add a sphere to the scene, returns the object
+    sphere_obj = rigid_obj_mgr.add_object_by_id(sphere_template_id)
+    # move sphere
+    sphere_obj.translation = np.array([2.50, 0, 0.2])
+
+    # simulate
+    observations = simulate(sim, dt=1.5, get_frames=make_video)
+
+    if make_video:
+        vut.make_video(
+            observations,
+            "rgba_camera_1stperson",
+            "color",
+            output_path + "sim_basics",
+            open_vid=show_video,
+        )
+
+    # [/basics]
+    rigid_obj_mgr.remove_all_objects()
+    # %%
+    # [dynamic_control]
+
+    observations = []
+    obj_templates_mgr.load_configs(str(os.path.join(data_path, "objects")))
+    # search for an object template by key sub-string
+    cheezit_template_handle = obj_templates_mgr.get_template_handles(
+        "data/objects/cheezit"
+    )[0]
+    # build multiple object initial positions
+    box_positions = [
+        np.array([2.39, -0.37, 0]),
+        np.array([2.39, -0.64, 0]),
+        np.array([2.39, -0.91, 0]),
+        np.array([2.39, -0.64, -0.22]),
+        np.array([2.39, -0.64, 0.22]),
+    ]
+    box_orientation = mn.Quaternion.rotation(
+        mn.Rad(math.pi / 2.0), np.array([-1.0, 0, 0])
+    )
+    # instance and place the boxes
+    boxes = []
+    for b in range(len(box_positions)):
+        boxes.append(rigid_obj_mgr.add_object_by_handle(cheezit_template_handle))
+        boxes[b].translation = box_positions[b]
+        boxes[b].rotation = box_orientation
+
+    # get the object's initialization attributes (all boxes initialized with same mass)
+    object_init_template = boxes[0].creation_attributes
+    # anti-gravity force f=m(-g)
+    anti_grav_force = -1.0 * sim.get_gravity() * object_init_template.mass
+
+    # throw a sphere at the boxes from the agent position
+    sphere_template = obj_templates_mgr.get_template_by_ID(sphere_template_id)
+    sphere_template.scale = np.array([0.5, 0.5, 0.5])
+
+    obj_templates_mgr.register_template(sphere_template)
```

Comment:
To me it seems like this should create a new template internally.  i.e. I would have expected
```
scaled_sphere_template_id = obj_templates_mgr.register_template(sphere_template)
```

Perhaps `update_template`?

Reply:
@jturner65 Correct me if I'm wrong here. I think it serves a dual purpose, right? You can get a copy of an attributes template (or a fresh one), modify it, and then register it with a new handle as a new cached template. Alternatively you can just update an existing attributes template using the default arguments. Do we want to separate these? In my mind this is very similar to a C++ std::map [] operator.

Reply:
I'd rather not separate them - that would be a fairly substantial refactor to the managed container.  All the functionality is accomplished currently based on context.